### PR TITLE
Validator Specific Profile Page Design & Integration

### PR DESCRIPTION
**Pull request by nblogist**
## Description
Integrated designs, charts, relevant APIs in the validator specific profile page, there are a few adjustments remaining (few design tweaks,  some data fields in validator_header_profile)

## Motivation and Context
- Pages over specific validators statistics; Pages for global statistics such as total number of offences/slashes/rewards over certain intervals #289 
- Design UI to show slashes statistics #293 
- Design UI to show rewards statistics #291 
- Design UI to show offences statistics. #290

## How has this been tested?
Tested with edgeware

## Have proper tags been added (for bug, enhancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [ ] yes, and they are tested: [enter the % coverage here]
- [ ] yes, and they are not tested: [enter the % coverage here]
- [ ] yes, but I did not run tests
- [x] no

### CODE REVIEW

File: client/scripts/views/pages/profile/validator_header_stats.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  getObservables: (attrs) => ({
+    groupKey: app.chain.class.toString(),
+    validators: (app.chain.base === ChainBase.Substrate) ? (app.chain as Substrate).staking.validators : null,
+    lastHeader: (app.chain.base === ChainBase.Substrate)
+      ? (app.chain as Substrate).staking.lastHeader
+      : null,
+  }),
+  view: (vnode) => {
+    const { account } = vnode.attrs;
+    const validators = (vnode.state.dynamic && vnode.state.dynamic.validators !== undefined) ? vnode.state.dynamic.validators : undefined;
```
Comment (drewstone) [inline client/scripts/views/pages/profile/validator_header_stats.ts:81]: ternary should be on separate lines:
```
let x = (condition)
  ? first
  : second;
```

File: client/scripts/views/pages/profile/validator_header_stats.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    const validatorStatInfo: IValidatorStatsInfo = validators ? (validators[vnode.attrs.address] ? {
```
Comment (drewstone) [inline client/scripts/views/pages/profile/validator_header_stats.ts:82]: looks like you can combine this to `(validators && validators[vnode.attrs.address])`

File: client/scripts/views/pages/profile/validator_profile_stats.ts [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+      ? (app.chain as Substrate).staking.lastHeader
+      : null,
+    annualPercentRate: (app.chain.base === ChainBase.Substrate)
+      ? (app.chain as Substrate).staking.annualPercentRate
+      : null
+  }),
+  oncreate: async (vnode) => {
+    m.request({
+      method: 'GET',
+      url: '/api/getSlashes',
```
Comment (drewstone) [inline client/scripts/views/pages/profile/validator_profile_stats.ts:79]: Please follow the structure of the rest of the repo, an example is below even though there are plenty of examples of how to construct the proper route.
```
`${app.serverUrl()}/createComment`
```

File: client/scripts/views/pages/profile/validator_profile_stats.ts [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+      vnode.state.fullYearSlashes = [];
+      if (response) {
+        vnode.state.fullYearSlashes = response.result.slashes;
+      }
+    }).catch((e: any) => {
+      vnode.state.fullYearSlashes = [];
+    });
+    m.request({
+      method: 'GET',
+      url: '/api/getSlashes',
```
Comment (drewstone) [inline client/scripts/views/pages/profile/validator_profile_stats.ts:91]: same here

**Review by drewstone — CHANGES_REQUESTED** [review]
Looks good, please add more comments and give some more insight into why so much is commented out and whether its meant to be removed or uncommented.

File: client/scripts/views/pages/profile/validator_header_stats.ts (hunk 1 above)
Comment (nblogist) [reply client/scripts/views/pages/profile/validator_header_stats.ts:81]: Done

File: client/scripts/views/pages/profile/validator_profile_stats.ts (hunk 3 above)
Comment (nblogist) [reply client/scripts/views/pages/profile/validator_profile_stats.ts:76]: Thanks for pointing the inconsistency out @drewstone !

File: client/scripts/views/pages/profile/validator_profile_stats.ts (hunk 4 above)
Comment (nblogist) [reply client/scripts/views/pages/profile/validator_profile_stats.ts:88]: Done

**Comment by drewstone** [comment]
Looks like a lot of re-formatting that should. not have been added as been added and a lot more of entire files commented out. I'd revert the formatting changes. Merge conflicts should be fixed too, what is `validators1` page also? That does not seem to fit.

**Comment by drewstone** [comment]
@YameenMalik is this PR going to be fixed up or should I close?

**Comment by YameenMalik** [comment]
> @YameenMalik is this PR going to be fixed up or should I close?
@drewstone you may close it as this and many other issues have been taken care over in this PR: https://github.com/hicommonwealth/commonwealth/pull/846

**Pull request closed without merging**